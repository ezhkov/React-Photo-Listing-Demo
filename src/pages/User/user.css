.user {
    margin: 50px auto;
    max-width: 300px;
}
.user .ant-card-body {
    position: relative;
    z-index: 1;
    text-align: center;
}
.user-card__photo-wrap {
    display: flex;
    justify-content: center;
    margin-top: -45px;
}
.user-card__title {
    font-size: 18px;
    margin: 10px 0;
}
.user-card {
    padding: 20px;
}
.user-card__btn-go-back {
    margin-top: 10px;
}
.user-card__first-img {
    width: 100%;
    height: auto;
    position: relative;
    z-index: 0;
}
.user-card__photo {
    width: 130px;
    margin: -50px auto 0;
    border-radius: 100%;
    z-index: 2;
}