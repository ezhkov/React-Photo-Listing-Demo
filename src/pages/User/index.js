import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  Row,
  Col,
  Avatar,
  Tabs,
} from 'antd';
import { userRequestAction } from '../../actions';
import {
  Spinner,
  UserStatistic,
  Error,
  SmallPhotoListing,
} from '../../components';
import './user.css';

const { TabPane } = Tabs;

class User extends PureComponent {
  componentDidMount = () => {
    const { match, userRequestAction: requestAction } = this.props;
    requestAction(match);
  };

  render() {
    const {
      match,
      isUserFetching,
      userPhoto,
      userFirstPhoto,
      requestError,
    } = this.props;
    return (
      <div className="user-container">
        { isUserFetching && (<Spinner />)}
        { !isUserFetching && !requestError && (
          <div>
            <Row style={{ display: 'flex', justifyContent: 'center' }}>
              <Col>
                <img
                  className="user-card__first-img"
                  alt="example"
                  src={userFirstPhoto}
                />
                <div className="user-card__photo-wrap">
                  <Avatar size={150} src={userPhoto} />
                </div>
              </Col>
            </Row>
            <Row
              style={{
                display: 'flex',
                justifyContent: 'center',
                overflow: 'hidden',
              }}
            >
              <Col>
                <div className="user-tabs">
                  <Tabs
                    className="user-tabs__item"
                    defaultActiveKey="1"
                  >
                    <TabPane
                      tab="My photos"
                      key="1"
                      className="user-tabs__pane"
                    >
                      <SmallPhotoListing
                        userId={match.params.id}
                        itemNum={0}
                        name="photos"
                      />
                    </TabPane>
                    <TabPane
                      tab="My likes"
                      key="2"
                      className="user-tabs__pane"
                    >
                      <SmallPhotoListing
                        userId={match.params.id}
                        itemNum={1}
                        name="likes"
                      />
                    </TabPane>
                    <TabPane
                      tab="My statistic"
                      key="3"
                      className="user-tabs__pane"
                    >
                      <UserStatistic userId={match.params.id} />
                    </TabPane>
                  </Tabs>
                </div>
              </Col>
            </Row>
          </div>
        )}
        { !isUserFetching && requestError && (
        <Error text="Sorry, user is not found" />
        )}
      </div>
    );
  }
}

User.propTypes = {
  isUserFetching: PropTypes.bool,
  userPhoto: PropTypes.string,
  userFirstPhoto: PropTypes.string,
  userRequestAction: PropTypes.func,
  requestError: PropTypes.bool,
  history: PropTypes.shape({
    prop: PropTypes.string,
  }),
  match: PropTypes.shape({
    prop: PropTypes.string,
  }),
};
User.defaultProps = {
  isUserFetching: true,
  userPhoto: '',
  userFirstPhoto: '',
  userRequestAction: () => {},
  requestError: false,
  history: {},
  match: {},
};

const mapStateToProps = (state) => {
  const { user } = state;
  return user;
};

const mapDispatchToProps = ({
  userRequestAction,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(User);
