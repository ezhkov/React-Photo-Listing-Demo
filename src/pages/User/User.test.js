import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Link, Route } from 'react-router-dom';
import User from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of User', () => {
  it('Initial', () => { 
    const props = {
      match: {
        params: {
          id: 'sC-BXbi9ajw',
        }
      }
    }

    const user = shallow(<User {...props} />);
    user.setState({ 
      userName: 'User111', 
      userLastName: 'UserLastName',
      userPortfolioUrl: 'userPortfolioUrl',
      userFirstPhoto: 'userFirstPhotoSrc',
      userPhoto: 'userPhotoSrc'
    }); 
    //expect(user.find('.user-card__title').text()).toEqual("User111 UserLastName");
    //expect(user.find('.user-card__first-img').prop('src')).toEqual("userFirstPhotoSrc");
    //expect(user.find('.user-card__photo').prop('src')).toEqual("userPhotoSrc");
    //expect(user.find('.user-card__link-portfolio').prop('href')).toEqual("userPortfolioUrl");
    //expect(user).toMatchSnapshot();
  });
});
