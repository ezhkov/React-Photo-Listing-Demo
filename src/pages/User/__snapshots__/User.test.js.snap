// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of User Test User snapshot 1`] = `ShallowWrapper {}`;
